--- pugixml/src/pugiconfig.hpp.orig	2014-11-25 05:17:23.000000000 +0100
+++ pugixml/src/pugiconfig.hpp	2015-03-13 14:31:03.185576918 +0100
@@ -38,8 +38,8 @@
 // #define PUGIXML_MEMORY_XPATH_PAGE_SIZE 4096
 
 // Uncomment this to switch to header-only version
-// #define PUGIXML_HEADER_ONLY
-// #include "pugixml.cpp"
+#define PUGIXML_HEADER_ONLY
+#include "pugixml.cpp"
 
 // Uncomment this to enable long long support
 // #define PUGIXML_HAS_LONG_LONG
