--- a/dvb.c	2012-04-06 06:27:14.000000000 +0800
+++ b/dvb.c	2014-11-04 16:09:24.930398918 +0800
@@ -56,7 +56,8 @@
 #define DVR_READ_TIMEOUT 30000000 /* 30 s */
 #define CA_POLL_PERIOD 100000 /* 100 ms */
 #define MAX_READ_ONCE 50
-#define DVR_BUFFER_SIZE 40*188*1024 /* bytes */
+//#define DVR_BUFFER_SIZE 40*188*1024 /* bytes */
+#define DVR_BUFFER_SIZE 10*188*1024 /* bytes */
 
 static int i_frontend, i_dvr;
 static fe_status_t i_last_status;
