--- vlc-2.2.4/configure.ac.org	2016-09-30 17:30:58.422945255 +0200
+++ vlc-2.2.4/configure.ac	2016-09-30 17:31:31.902945831 +0200
@@ -2323,10 +2323,10 @@
 [  --enable-avcodec        libavcodec codec (default enabled)])
 AS_IF([test "${enable_avcodec}" != "no"], [
   PKG_CHECK_MODULES(AVCODEC,[libavcodec >= 53.34.0 libavutil >= 51.22.0], [
-    PKG_CHECK_EXISTS([libavutil < 55],, [
-      AC_MSG_ERROR([libavutil versions 55 and later are not supported.])
-    ])
-    VLC_SAVE_FLAGS
+#    PKG_CHECK_EXISTS([libavutil < 55],, [
+#      AC_MSG_ERROR([libavutil versions 55 and later are not supported.])
+#    ])
+#    VLC_SAVE_FLAGS
     CPPFLAGS="${CPPFLAGS} ${AVCODEC_CFLAGS}"
     CFLAGS="${CFLAGS} ${AVCODEC_CFLAGS}"
     AC_CHECK_HEADERS(libavcodec/avcodec.h)
